pennylane/ops/op_math/composite.py

```diff
@@ -55,6 +57,7 @@ def __init__(
         self._wires = qml.wires.Wires.all_wires([op.wires for op in operands])
         self._hash = None
         self._has_overlapping_wires = None
```

Comment:
With the introduction of `self.overlapping_ops` does `self._has_overlapping_wires` become useless? i.e is it called/used anywhere else?

Reply:
Nope! Deleting property...

Reply:
Yes! In the `self.has_diagonalizing_gates` property and the `Prod.sparse_matrix` method.